packages/wrangler/src/pages/build.ts

```diff
@@ -483,22 +518,54 @@ We looked for the Functions directory (${basename(
 		);
 	}
 
+	// The build metadata for this build (a hash of the config file it was based on, and the output directory that was used) is usually generated from wrangler.toml
+	// However, frameworks may want to allow users to use a completely different config file, and so may want to fully generate a `_build-metadata.json` file themselves
+	// This is an advanced use case, and should go along with a custom `_worker.bundle` file. The vast majority of users won't need to cutomise this!
```

Comment:
would it make sense here to error if a custom `_build-metadata.json` file is used but no custom `_worker.bundle` file is? (e.g. `if (existsSync(prospectiveBuildMetadataPath) && !foundWorkerBundle) throw`)